client/my-sites/stats/stats-purchase/stats-purchase-checkout-redirect.tsx

```diff
@@ -24,9 +24,13 @@ const getStatsCheckoutURL = (
 	adminUrl?: string,
 	isUpgrade?: boolean
 ) => {
-	const isFromJetpack = from?.startsWith( 'jetpack' );
-	// Get the checkout URL for the product, or the siteless checkout URL if from Jetpack or no siteSlug is provided
-	const checkoutType = ( isFromJetpack && ! isUpgrade ) || ! siteSlug ? 'jetpack' : siteSlug;
+	// Get the checkout URL for the product.
+	// The following cases get the logged out flow:
+	// - If the user is coming from My Jetpack
+	// - If no siteSlug is provided
+	// - If it's not an upgrade (currently causes errors in the logged in flow)
```

Comment:
```suggestion
	// - If it's not an upgrade (currently causes errors in the logged out flow)
```
Is it [causing multiple purchases](https://github.com/Automattic/wp-calypso/pull/91661) with the site-only (logged out) flow?

Reply:
If the site has a PWYW plan then yes. The purchase completes but assignment to the site fails and Store Admin shows the site as having an unassigned license.

If the current plan is not detected (for example, adding a 10k plan on top of a 10k plan), the cart shows an error and prevents the checkout.